ls1intum/Artemis
# [Feature] Side-by-side view for plagiarism inspection (Modeling Exercises)

### Checklist
- [x] Client: I added screenshots/screencasts of my UI changes

### Motivation and Context
For motivation and context, check out #2100.

### Description
I added the `ArtemisPlagiarismModule` under `exercises/shared` which exports the `PlagiarismInspectorComponent`.

To implement the `PlagiarismSplitViewComponent`, I added the [split.js](https://split.js.org/) library.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration > Your Course > Exercises
3. Create a modeling exercise
4. Use multiple accounts to create similar submissions for your modeling exercise
5. Go to the exercise details page
6. Click 'Check plagiarism'

You should now be redirected to a new page with the plagiarism inspector. Check the following functionality:

1. Given similar submissions exist, clicking on 'Check Plagiarism' should fetch detected plagiarisms and display them in the sidebar on the left
2. Detected plagiarisms are sorted by their similarity in descending order
2. Selecting a plagiarism item in the sidebar loads the related submissions and displays them in the split view
3. The split view should be resizable
4. Split view shortcuts should work (top right)
5. Confirming/Denying a plagiarism should update its status indicator in the sidebar
6. Downloading the results should include a 'confirmed' column/attribute if the plagiarism was confirmed/denied (both JSON and CSV)

### Screenshots
<img width="1680" alt="Screenshot 2020-09-28 at 15 30 08" src="https://user-images.githubusercontent.com/21973908/94447796-7db4c880-01aa-11eb-95d6-048f9547d314.png">
